Automattic/wp-calypso
# Refactor the DailyBackupStatus component and add unit tests

#### Changes proposed in this Pull Request

* Refactor the `DailyBackupStatus` component to more clearly reflect what it's doing
* Create separate component files for each of the possible statuses
* Clean up styles for `DailyBackupStatus`, removing unused rules and moving rules for the statuses into the corresponding directory (`status-card`)
* Add unit tests to cover logic that determines which backup status to show

#### Testing instructions

##### A short preamble

This pull request needs a full battery of regression tests. Ideally, we want to cover:
* five possible backup statuses;
* on both mobile and desktop layouts;
* on both Jetpack.com and WordPress.com;
* for both Daily and Real-time backups.

This multiplies out to 40 total combinations, though, so it could take a bit. It may be more efficient for each reviewer to comment with a description of which cases they were able to test (e.g., all Jetpack Cloud cases, all mobile cases, etc.), noting any failures. That way, the load can be cleanly and simply divided.

**Tip:** React DevTools allows you to change component properties, meaning you can simulate different backup statuses without actually creating the necessary conditions by hand. For example, changing the `selectedDate` property can make the `DailyBackupStatus` component behave as if we're looking at today, or any other date.

##### Specific things to test for on the main Backup page

* Check that all unit tests pass (`yarn test-client`)
* Check that date navigation still works properly
* Check that all pages load correctly with no console errors
* Check that all dates appear correct
* Check that all margins and padding look correct and mirror production
* Check that all fonts and colors look correct and mirror production

#### Reference screenshots

<img width="382" alt="Screen Shot 2020-08-31 at 10 47 06" src="https://user-images.githubusercontent.com/670067/91739879-2524e800-eb78-11ea-893f-53e1f060d924.png">

<img width="382" alt="Screen Shot 2020-08-31 at 10 47 28" src="https://user-images.githubusercontent.com/670067/91739880-26561500-eb78-11ea-8b26-e31de59ce495.png">

<img width="385" alt="Screen Shot 2020-08-31 at 10 47 38" src="https://user-images.githubusercontent.com/670067/91739883-26eeab80-eb78-11ea-9e4a-caf8a6babd29.png">

<img width="383" alt="Screen Shot 2020-08-31 at 10 48 27" src="https://user-images.githubusercontent.com/670067/91739885-27874200-eb78-11ea-94de-e1e1494732c2.png">

<img width="384" alt="Screen Shot 2020-08-31 at 10 52 09" src="https://user-images.githubusercontent.com/670067/91739886-27874200-eb78-11ea-99ff-838255ad49db.png">
